packages/Tag/src/Tag.styles.js

```diff
@@ -0,0 +1,165 @@
+import styled, { css } from "styled-components";
+import tokens from "@paprika/tokens";
+import RawButton from "@paprika/raw-button";
+import { fontSize } from "@paprika/stylers/lib/helpers";
+import { truncateText } from "@paprika/stylers/lib/includes";
+
+const tagColorStyles = {
+  black: css`
+    background: ${tokens.color.black};
+  `,
+
+  blue: css`
+    background: ${tokens.color.blue};
+  `,
+
+  grey: css`
+    background: ${tokens.color.blackLighten70};
+    color: ${tokens.color.black};
+  `,
+
+  green: css`
+    background: ${tokens.color.greenDarken10};
+  `,
+
+  orange: css`
+    background: ${tokens.color.orangeDarken10};
+  `,
+
+  noRisk: css`
+    background: ${tokens.color.blackLighten70};
+    color: ${tokens.color.black};
+  `,
+
+  lightBlue: css`
+    background: ${tokens.color.blueLighten50};
+    color: ${tokens.color.blueDarken20};
+  `,
+
+  lightOrange: css`
+    background: ${tokens.color.orangeLighten40};
+    color: ${tokens.color.orangeDarken20};
+  `,
+
+  lowRisk: css`
+    background: #299a7a;
+  `,
+
+  mediumRisk: css`
+    background: #c9af28;
+  `,
+
+  highRisk: css`
+    background: #cd3c44;
+  `,
+
+  alert: css`
+    background: none;
+    color: ${tokens.color.orangeDarken10};
+  `,
+};
+
+export const Tags = styled.ul(() => {
+  return css`
+    display: flex;
+    flex-wrap: wrap;
+    line-height: 1;
+    margin: 0;
+    padding: 0;
+  `;
+});
+
+const borderColorStyles = ({ borderColor }) => css`
+  border: 1px solid ${borderColor};
+`;
+
+const tagStyles = ({ tagColor, borderColor }) => css`
+  align-items: center;
+  background: ${tokens.color.blackLighten70};
+  ${borderColor ? borderColorStyles : ""}
+  border-radius: ${tokens.space[0] * 2}px;
+  color: ${tokens.color.white};
+  display: flex;
+  line-height: 1;
+  margin-bottom: 2px;
+  margin-right: ${tokens.spaceSm};
+  max-width: 100%;
+  padding: 2px;
+  ${tagColorStyles[tagColor]}
+
+  [data-pka-anchor="avatar"] {
+    margin-right: ${tokens.space};
+  }
+`;
+
+export const RawButtonTag = styled(RawButton)(tagStyles);
+
+export const Tag = styled.li(tagStyles);
+
+export const Ellipsis = styled.div(({ isChildString, size }) => {
+  const getFontSize = {
+    medium: fontSize(-1),
+    large: fontSize(0),
+  };
+
+  const paddingGap = {
+    medium: `${tokens.space[0] / 2}px`,
+    large: tokens.space,
+  };
+
+  const padding = isChildString
+    ? `padding: 0 ${paddingGap[size]} 1px ${paddingGap[size]};`
+    : `padding-right: ${paddingGap[size]}`;
+
+  // todo: fix truncateText not showing ... with display inline-flex, only with display: block....
```

Comment:
Truncation is not working properly, as is not showing the `...` without `display: block`. When using `display:block` however the text content in the Tag was being vertically aligned incorrectly by the font size of text content in the avatar beside it.

Reply:
I had the same problem in a component that I'm currently working on, what I did is make use of two divs one that wraps the flex vertical attribute, and another one `nested` that use display block, here you can see the example that I think is similar to your issue https://github.com/acl-services/paprika/pull/939/files#diff-09d3038e83adb04da25396535ba5f213c0d69ea92fc68027ea33e0bbb7002c0dR3

Reply:
Ok, I discussed with Mike, I'm passing the avatar in as a prop now and using `display:block` on the tag content, so ellipse works now and i think is good for the api